.led/plugins/php.plugin.sh

```diff
@@ -123,33 +129,102 @@ php_xdebug()
         enable) xdebug_value=1;;
         disable) xdebug_value=0;;
         switch)
-          xdebug_value=$(${VENDORS_BIN_JYPARSER} "${dc_file}" get "${dc_key}")
+          xdebug_value=$(${VENDORS_BIN_JYPARSER} "${dc_file}" get "${dc_xdebug_key}")
           [ "${xdebug_value}" == "null" ] && xdebug_value=0
           # value can be 1 or 0, so substract to reverse
           xdebug_value=$((1 - xdebug_value))
           ;;
     esac
 
-
+    local dc_xdebug_value=$(${VENDORS_BIN_JYPARSER} "${dc_file}" get "${dc_xdebug_key}")
     if [ ${xdebug_value} -eq 0 ]; then
-      ${VENDORS_BIN_JYPARSER} "${dc_file}" del "${dc_key}" > "${dc_file_tmp}" \
+      _check_xdebug_state ${dc_xdebug_value} ${xdebug_value}
+
+      ${VENDORS_BIN_JYPARSER} "${dc_file}" del "${dc_xdebug_key}" > "${dc_file_tmp}" \
       && mv "${dc_file_tmp}" "${dc_file}"
+      _disable_debugger
       ret=$?
       message="XDebug disabled"
     else
-      ${VENDORS_BIN_JYPARSER} "${dc_file}" set "${dc_key}" 1 > "${dc_file_tmp}" \
+      _check_xdebug_state ${dc_xdebug_value} ${xdebug_value}
+
+      ${VENDORS_BIN_JYPARSER} "${dc_file}" set "${dc_xdebug_key}" 1 > "${dc_file_tmp}" \
       && mv "${dc_file_tmp}" "${dc_file}"
+      _enable_debugger
       ret=$?
       message="XDebug enabled"
     fi
 
     # refresh container
     if [ $ret -eq 0 ]; then
       echo "${message}"
-      $0 up
+#      $0 up
     fi
+}
 
+_enable_debugger()
+{
+    local phpstorm_ip
+    local plugin_dir="$(_plugin_files_dir "php")"
+    local origin_xdebug_config_file="${plugin_dir}/90-xdebug-config.ini"
+    local dc_volumes_key=$(_php_get_service_key "volumes")
+    local dc_ide_config_key=$(_php_get_service_key "environment" "PHP_IDE_CONFIG")
+    local dc_volumes_length
+    local dc_xdebug_env_value
+
+    if [[ -f "${origin_xdebug_config_file}" ]]; then
+      target_xdebug_config=".led/90-xdebug-config.ini"
```

Comment:
you must create `.led` directory if missing, otherwise the `sed` command can't create output file